gobblin-service/src/main/java/org/apache/gobblin/service/modules/utils/FlowCompilationValidationHelper.java

```diff
@@ -216,6 +217,12 @@ static boolean isFlowRunning(String flowGroup, String flowName, DagManagementSta
     }
 
     for (FlowStatus flowStatus : flowStatusList) {
+      if (flowStatus.getFlowExecutionId() == flowExecutionId) {
+        // a duplicate call to this method indicate that the prior caller of this method could not complete the required action,
+        // so we ignore any flow status for the current execution to give the caller another chance to complete them
+        continue;
```

Comment:
given we expect re-attempts to be rare, let's at least log this condition.  we really expect it only in `COMPILED` state, so let's do `.info` level.  but if it's any other status, that merits `.warn`

perhaps if the state is actually `RUNNING` it should even be `.error` level

Reply:
updated